### ISSUE TITLE
[Bug]: Resource aws_lambda_invocation does not propagate changes to result attribute when changing input, using 'CRUD' lifecycle scope

### ISSUE THREAD

**Issue by dp-152** [hashicorp/terraform-provider-aws#34262]
### Terraform Core Version

1.6.3

### AWS Provider Version

5.24.0

### Affected Resource(s)

aws_lambda_invocation

### Expected Behavior

Resource should invalidate the 'result' attribute when changing input parameters.

### Actual Behavior

Result is not invalidated on input change, and dependent resources cannot plan for result changes, causing a state inconsistency after an apply. Any failed postcondition checks on the resource that are bound to the result are not re-evaluated, forcing the user to manually taint the resource.

### Relevant Error/Panic Output Snippet

```shell
data.archive_file.function: Reading...
data.archive_file.function: Read complete after 0s [id=7ca40895cbd79f316b62feca568132a616724ba1]
data.aws_iam_policy_document.this: Reading...
data.aws_iam_policy_document.this: Read complete after 0s [id=2690255455]
aws_iam_role.this: Refreshing state... [id=terraform-crud-invocation-test-lambda-role]
aws_iam_role_policy_attachment.this: Refreshing state... [id=terraform-crud-invocation-test-lambda-role-20231104062651665700000001]
aws_lambda_function.this: Refreshing state... [id=terraform-crud-invocation-test]
aws_lambda_invocation.this: Refreshing state... [id=terraform-crud-invocation-test_$LATEST_ada7d1b1e098537927118ec3c49338fa]
aws_ssm_parameter.generated: Refreshing state... [id=/tf/lambda-invocation-result/server-generated]
aws_ssm_parameter.prefixed: Refreshing state... [id=/tf/lambda-invocation-result/prefixed-username]
aws_ssm_parameter.suffixed: Refreshing state... [id=/tf/lambda-invocation-result/suffixed-username]

Terraform used the selected providers to generate the following execution plan. Resource actions are indicated with the following symbols:
  ~ update in-place

Terraform will perform the following actions:

  # aws_lambda_invocation.this will be updated in-place
  ~ resource "aws_lambda_invocation" "this" {
        id              = "terraform-crud-invocation-test_$LATEST_ada7d1b1e098537927118ec3c49338fa"
      ~ input           = jsonencode(
            {
              - prefix   = "service"
              - username = "usr"
            }
        ) -> (known after apply)
        # (5 unchanged attributes hidden)
    }

  # aws_ssm_parameter.generated will be updated in-place
  ~ resource "aws_ssm_parameter" "generated" {
        id             = "/tf/lambda-invocation-result/server-generated"
      + insecure_value = (known after apply)
        name           = "/tf/lambda-invocation-result/server-generated"
        tags           = {}
      ~ value          = (sensitive value)
      ~ version        = 1 -> (known after apply)
        # (5 unchanged attributes hidden)
    }

  # aws_ssm_parameter.prefixed will be updated in-place
  ~ resource "aws_ssm_parameter" "prefixed" {
        id             = "/tf/lambda-invocation-result/prefixed-username"
      + insecure_value = (known after apply)
        name           = "/tf/lambda-invocation-result/prefixed-username"
        tags           = {}
      ~ value          = (sensitive value)
      ~ version        = 1 -> (known after apply)
        # (5 unchanged attributes hidden)
    }

Plan: 0 to add, 3 to change, 0 to destroy.
aws_lambda_invocation.this: Modifying... [id=terraform-crud-invocation-test_$LATEST_ada7d1b1e098537927118ec3c49338fa]
aws_lambda_invocation.this: Modifications complete after 1s [id=terraform-crud-invocation-test_$LATEST_25ae9ea34abc579c5eb6227d576a862d]
╷
│ Error: Provider produced inconsistent final plan
│ 
│ When expanding the plan for aws_ssm_parameter.generated to include new values learned so far during apply, provider "registry.terraform.io/hashicorp/aws" produced an invalid new value for .value: inconsistent
│ values for sensitive attribute.
│ 
│ This is a bug in the provider, which should be reported in the provider's own issue tracker.
╵
╷
│ Error: Provider produced inconsistent final plan
│ 
│ When expanding the plan for aws_ssm_parameter.prefixed to include new values learned so far during apply, provider "registry.terraform.io/hashicorp/aws" produced an invalid new value for .value: inconsistent
│ values for sensitive attribute.
│ 
│ This is a bug in the provider, which should be reported in the provider's own issue tracker.
```

### Terraform Configuration Files

**[\[Gist\]](https://gist.github.com/dp-152/b6ab8b806a2db91633b63a5add2e3a3c)**
Minimal working example - provisions a lambda function and required IAM role, an invocation resource for the created function, and 3 SSM parameters which depend on the invocation output.

### Steps to Reproduce

**Test case 1: Final plan inconsistency**
- Apply the attached configuration: The configuration contains a aws_lambda_invocation resource and 3 SSM parameters depending on the invocation result. Apply should occur without any errors
- Change one of the inputs to a different, valid string, and apply: The plan output will not show any changes to the SSM parameters, but the apply will run without issues.
- Revert the changed parameter to the original value or any other valid string, then apply again: The error should occur here, with a message `Provider produced inconsistent final plan` for every SSM parameter affected.

**Test case 2: Postcondition lock**
- Apply the attached configuration from a clean state.
- Set the 'username' field on the invocation input to "fail" and apply again: This will cause the function to return "false" on the "success" field, causing the postcondition to fail.
- Change the 'username' field back to a valid value, then attempt to apply again: This will fail with the same postcondition check, as the resource will not propagate any change to the result. The resource is now locked in this state, unless it is tainted manually

### Debug Output

Test case 1 - Inconsistent final state:

[test-case-1-inconsistent-plan.gpgencrypted.zip](https://github.com/hashicorp/terraform-provider-aws/files/13256041/test-case-1-inconsistent-plan.gpgencrypted.zip)

Test case 2 - Postcondition lock:

[test-case-2-postcondition-lock.gpgencrypted.zip](https://github.com/hashicorp/terraform-provider-aws/files/13256042/test-case-2-postcondition-lock.gpgencrypted.zip)

### Important Factoids

The issue seems to stem from [internal/service/lambda/invocation.go:72-75](https://github.com/hashicorp/terraform-provider-aws/blob/e4526080946e63dd40ef99d6b22459cbdebb33fe/internal/service/lambda/invocation.go#L72C3-L75C5). There is no diff checker registered to invalidate the "result" attribute when "input" is changed.

My use case involves invoking a lambda function that provisions and manages PostgreSQL database resources (tables, roles, functions, etc), and the data from the invocation result is to be used in other resources that depend on them (I am aware of [cyrilgdn/postgresql](https://github.com/cyrilgdn/terraform-provider-postgresql), but it doesn't suit my particular use case). My current workaround is loosely matching the expected output with local variables, and using those instead of the resource output directly, but this gets extremely difficult to manage.

### Would you like to implement a fix?

Yes

**Comment by dp-152** [comment]
Proposed fix: #34263 [WIP]

**Comment by mizael-fitdance** [comment]
Can confirm no longer reproducible after provider v5.82.0.
